### PR TITLE
Honor EMCC_CORES in test runner's `build_library`

### PR DESCRIPTION
**Pull request by sbc100**
This is important on circleci where the machines report a lot of
cores but we don't want to actually use that many (we force the
core count 4 there)..

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -367,6 +367,8 @@ jobs:
     executor: bionic
     steps:
       - checkout
+      - run: pip2 install flake8==3.7.8
+      - run: pip3 install flake8==3.7.8
```
Comment (kripken) [inline .circleci/config.yml:371]: looks unrelated?

**Comment by kripken** [comment]
Do those bots report more than 4 cores? I thought they only see the virtual cores they get, which is a small number. But maybe I don't remember right.

**Comment by sbc100** [comment]
> Do those bots report more than 4 cores? I thought they only see the virtual cores they get, which is a small number. But maybe I don't remember right.

Yeah they seem to be reporting like 46 or some high number like that.

File: .circleci/config.yml (hunk 1 above)
Comment (sbc100) [reply .circleci/config.yml:371]: Yeah, I just noticed we don't pin a version here..  want me to split it out?

**Comment by sbc100** [comment]
If you look at this test run for example you can see "make -j36" appearing a bunch:

https://app.circleci.com/pipelines/github/emscripten-core/emscripten/6301/workflows/8a87479d-b2c3-4e94-bda5-901d33999488/jobs/326770/steps

**Comment by sbc100** [comment]
We explicitly set `EMCC_CORES: "4"` in circle CI config so we can control this.

File: .circleci/config.yml (hunk 1 above)
Comment (kripken) [reply .circleci/config.yml:371]: No, seems fine, I just wanted to make sure it's not unintended.

**Comment by kripken** [comment]
Wow, that is a lot of cores for a fairly small bot to report...
